pkg/controller/stack/stack_controller.go

```diff
@@ -656,7 +672,7 @@ func (sess *reconcileStackSession) SetupPulumiWorkdir(gitAuth *auto.GitAuth) err
 		return errors.Wrap(err, "installing project dependencies")
 	}
 
-	return nil
+	return err
```

Comment:
I can't quite grasp this change...

Reply:
> I can't quite grasp this change...

Since the defered function will check the value of `err`, the change makes sure returned value is through err. Could have done the same with a named return value. Happy to use that if that makes things clearer.

Reply:
> I can't quite grasp this change...

Its actually unnecessary. Removed.